oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticBulkProcessorHandler.java

```diff
@@ -161,12 +171,15 @@ public boolean close() {
             return false;
         }
 
-        try {
-            phaser.awaitAdvanceInterruptibly(phase, indexDefinition.bulkFlushIntervalMs * 5, TimeUnit.MILLISECONDS);
-        } catch (InterruptedException | TimeoutException e) {
-            LOG.error("Error waiting for bulk requests to return", e);
+        if (waitForESAcknowledgement) {
```

Comment:
I think that's OK. But what about the return value? Should we always return `false`? Or is it OK to return `true` when some `afterBulk` acknowledges a successful operation before closing the processor? (this can happen especially when the bulk processor has to send a lot of updates.

Reply:
We are returning  updatesMap.containsValue(Boolean.TRUE); 
If any one afterBulk would have had a success till now, this should be true correct ?

Reply:
Correct. In general, we would almost certainly return `true` when the bulk will process a lot of operations (> 250) because we will very likely receive some acknowledgment in the meantime. Instead, when the bulk processes a handful of operations we will very likely return false because we don't wait for the acknowledgment. Just wondering if this is correct or can cause any issues at the upper level.

Reply:
No there shouldn't be any issue as far as I can tell. This If block here wouldn't be executed if we return false https://github.com/apache/jackrabbit-oak/blob/trunk/oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/spi/editor/FulltextIndexEditorContext.java#L160 

This would just not update the last indexed time on the :status node, which would eventually be updated when we send true in the async cycle update (which would be after a few seconds).

Also, in case of small updates, when we do send false, I think that is correct, because we are unsure at that moment if the index is updated or not.